internal/consensus/state.go

```diff
@@ -2060,8 +2069,12 @@ func (cs *State) defaultSetProposal(proposal *types.Proposal, recvTime time.Time
 	}
 
 	proposal.Signature = p.Signature
-	cs.Proposal = proposal
-	cs.ProposalReceiveTime = recvTime
+	func() {
+		cs.mtx.Lock()
+		defer cs.mtx.RUnlock()
+		cs.Proposal = proposal
+		cs.ProposalReceiveTime = recvTime
+	}()
```

Comment:
This one would be fine to just bracket, there's no real panic risk.

```suggestion
	cs.mtx.Lock()
	cs.Proposal = proposal
	cs.ProposalReceiveTime = recvTime
	cs.mtx.Unlock()
```

N.B.: Whether or not you like this change, we should fix RUnlock → Unlock.